SpatialGDK/Source/SpatialGDK/Public/SpatialConstants.h

```diff
@@ -258,38 +258,69 @@ const FString ZoningAttribute = DefaultServerWorkerType.ToString();
 // A list of components clients require on top of any generated data components in order to handle non-authoritative actors correctly.
 const TArray<Worker_ComponentId> REQUIRED_COMPONENTS_FOR_NON_AUTH_CLIENT_INTEREST = TArray<Worker_ComponentId>
 {
+	// Actor components
 	UNREAL_METADATA_COMPONENT_ID,
 	SPAWN_DATA_COMPONENT_ID,
 	RPCS_ON_ENTITY_CREATION_ID,
+
+	// Multicast RPCs
 	MULTICAST_RPCS_COMPONENT_ID,
-	NETMULTICAST_RPCS_COMPONENT_ID_LEGACY
+	NETMULTICAST_RPCS_COMPONENT_ID_LEGACY,
+
+	// Global state components
```

Comment:
Are we using these result types for the one shot queries the client makes for the GSM? If not, then we don't need these right?

Reply:
haven't touched one shot queries- I'll test without these components here, but just to be sure, is there definitely nothing that might be changed in the near future that would mean clients need interest in global state? Having these components here should have negligible impact on performance as these components are rarely updated AFAIK, and in case any logic does require changing global state having these will avoid the components mysteriously existing if you do not have too much context on the interest set up

Could be better to change the one shot to instead use this result type in order to be consistent across the two?

Reply:
Looks like the one shot is just full snapshot and awkwardly self contained to the `GlobalStateManager` so would be a bigger change to modify it to use this result type. Will raise a ticket to clean up one shots with result types.

Reply:
[unr-2866](https://improbableio.atlassian.net/browse/UNR-2866)

Reply:
also saw some weird behaviour in the handover gym on trying this (client didn't possess the camera, or whatever the string of words is that means the camera moves to see the cubes) so I think this is going to take a little more work.

Reply:
I'm okay with leaving these in, so this array can work with the one shot query also.